gtsam/linear/JacobianFactor.cpp

```diff
@@ -491,10 +492,17 @@ bool JacobianFactor::equals(const GaussianFactor& f_, double tol) const {
 
 /* ************************************************************************* */
 Vector JacobianFactor::unweighted_error(const VectorValues& c) const {
-  Vector e = -getb();
-  for (size_t pos = 0; pos < size(); ++pos)
-    e += Ab_(pos) * c[keys_[pos]];
-  return e;
+  const Vector v = c.vector(keys_);
```

Comment:
We're still doing two mallocs here. Maybe inline vector(keys_) so you can avoid the v malloc? Maybe too small a win, but with 300k factors that's 300k mallocs saved.